### DISCUSSION TITLE
POST submissions not working

### DISCUSSION THREAD

**Discussion by anthonyfrancq** [PaperStrike/Pjax#98] [Q&A] [answered]
Hi there, I seem to be having a hard time getting this to work. I have included the JS file, do I need NodeJS installed for any of the other libraries it requires to work?

https://www.affwise.io/dashboard/login.php

**Comment by PaperStrike** [comment, marked as answer]
From the site you provided, I didn't find any other page to test. Pjax handles the switches between pages that have similar HTML structure, which means you should provide at least two page for me to debug your problem.

You don't need NodeJS to use this.

The pjax you using isn't the one in this repo, please turn to their repos, users, or authors to find your answers.

**Reply by anthonyfrancq** [reply]
Apologies, I was testing another Pjax repo, I added this one back on it seems to work for the link but for the form it does a full page load. 

https://www.affwise.io/dashboard/login.php - Click on Login.

I am using this for the configuration.

`const pjax = new Pjax({
  selectors: [
    'title',
    'body'
  ],
});`

**Reply by PaperStrike** [reply]
[The default trigger](https://github.com/PaperStrike/Pjax/tree/v1.1.0#defaulttrigger) does not support POST submissions yet. It has been scheduled to be shipped in ~1.2.0~ 2.0.0, but I can't promise when will finish.

Before this get supported, you may try to integrate with the [`loadURL`](https://github.com/PaperStrike/Pjax/tree/v1.1.0#loadurl) method. Basicly the steps are calling [`preventDefault`](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault) on the submit event, construct the form data using [`FormData`](https://developer.mozilla.org/en-US/docs/Web/API/FormData) API, send the POST with [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) or [`XMLHttpRequest`](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), and then use [`loadURL`](https://github.com/PaperStrike/Pjax/tree/v1.1.0#loadurl).

**Reply by anthonyfrancq** [reply]
If possible are you able to provide a quick example using what you described above?

**Reply by PaperStrike** [reply]
2.0.0 released now.
